usbpd/src/protocol_layer/message/request.rs

```diff
@@ -257,6 +258,27 @@ impl PowerSource {
         None
     }
 
+    /// Find a suitable PDO for a Programmable Power Supply (PPS) by evaluating the provided voltage
+    /// request against the source capabilities.
+    ///
+    /// Reports the index of the found PDO, and the augmented supply instance, or `None` if there is no match to the request.
+    fn find_pps_voltage(
+        source_capabilities: &pdo::SourceCapabilities,
+        voltage: ElectricPotential,
+    ) -> Option<(usize, &pdo::Augmented)> {
+        for (index, cap) in source_capabilities.pdos().iter().enumerate() {
+            if let pdo::PowerDataObject::Augmented(augmented) = cap {
+                if let Augmented::Spr(spr) = augmented {
+                    if spr.min_voltage() <= voltage && spr.max_voltage() >= voltage {
+                        return Some((index, augmented));
+                    }
+                }
+            }
+        }
```

Comment:
Can we unroll this a bit?

```suggestion
        for (index, cap) in source_capabilities.pdos().iter().enumerate() {
            let pdo::PowerDataObject::Augmented(augmented) = cap else {
                trace!("Skip non-augmented PDO {}", cap);
                continue;
            }
            
            // Handle EPR when supported.
            match augmented {
                Augmented::Spr(spr) => {
                    if spr.min_voltage() <= voltage && spr.max_voltage() >= voltage {
                        return Some((index, augmented));
                    }
                }
                _ => trace!("Skip PDO, only SPR is supported")
            }
```